env_config/load_profile.py

```diff
@@ -0,0 +1,56 @@
+"""
+This sample demonstrates loading a named environment configuration profile and
+overriding its values with environment variables.
+"""
+
+import asyncio
+from pathlib import Path
+
+from temporalio.client import Client
+from temporalio.envconfig import ClientConfig
+
+
+async def main():
+    """
+    Demonstrates loading a named profile and overriding it with env vars.
+    """
+    print("--- Loading 'staging' profile with environment variable overrides ---")
+
+    config_file = Path(__file__).parent / "config.toml"
+    profile_name = "staging"
+
+    # In a real application, these would be set in your shell or deployment
+    # environment (e.g., `export TEMPORAL_ADDRESS=localhost:7233`).
+    # For this sample, we pass them as a dictionary to demonstrate.
+    override_env = {
+        "TEMPORAL_ADDRESS": "localhost:7233",
+    }
```

Comment:
I don't think this is how we should encourage people to do overrides. Rather we should encourage them to alter the returned `connect_config`. They shouldn't mock environment variables.